bika/lims/browser/bika_listing.py

```diff
@@ -791,6 +728,99 @@ def _process_request(self):
+            # The value for sort_on is not an index, we need to force manual
+            # sorting so items will be sorted programmatically with python
+            self.manual_sort_on = self.sort_on
+
+        else:
+            # sort_on is an index, add it into the contentFilter so it can be
+            # used in the advanced query later
+            self.contentFilter['sort_on'] = self.sort_on
+
+        # By default, if sort_on is set, sort the items ASC
+        # Trick to allow 'descending' keyword instead of 'reverse'
+        sort_order = self.request.get(self.form_id + "_sort_order", None)
+        if not sort_order:
+            sort_order = self.contentFilter.get('sort_order', self.sort_order)
+        if sort_order != "ascending":
+            sort_order = "descending"
+        self.contentFilter['sort_order'] = sort_order
+        self.sort_order = sort_order
+
+    def get_sort_on(self):
+        """
+        Returns the value by which this list must be sorted on.
+        Returns the name of the index to be used for sorting by using an
+        advancedQuery if defined in self.columns. Otherwise, returns the name
+        of the column the list must be sorted on.
+        :return: the sort_on index or column name
+        :rtype: str
+        """
+
+        def corrected_sort_value(value):
+            """
+            Checks the value passed in against the columns and indexes.
+            If the value passed is a column name, will return the index of the
+            column if defined in self.columns. Otherwise, will return the
+            column name, but only if a column is defined in self.columns for
+            the value passed in. If non of both cases, returns None
+            """
+            if value in self.columns:
+                # The sort_on value refers to a column name. We try to get the index
+                # associated to this column name, if any
+                return self.columns[value].get('index', value)
+
+            if value in self.get_columns_indexes():
+                # The sort_on value refers to an index.
+                return value
+
+            # The sort_on value is neither an index nor a column. This should
+            # never happen
+            msg = "{}: sort_on is '{}', not a valid column/index".format(
+                self.__class__.__name__, value)
+            logger.warning(msg)
+            return None
+
+        # Request sort_on value has priority over contentsFilter's sort value,
+        # as well as self.sort_on (default sort_on criteria set for this list).
+        request_sort_on = self.request.get(self.form_id + "_sort_on", None)
+        filter_sort_on = self.contentFilter.get("sort_on", None)
+        sort_on_values = [request_sort_on, filter_sort_on, self.sort_on]
+        for sort_value in sort_on_values:
+            if not sort_value:
+                continue
+            if sort_value in (list, tuple):
+                sort_value = sort_value[0]
```

Comment:
Not sure when this happens, but this fails for an empty list/tuple with a KeyError

Reply:
Well spotted, I missed to evaluate the type against the list.

Reply:
better use `isinstance(sort_value, (list, tuple))` and then `len(sort_value)==1`